sys/kern/sched.c

```diff
@@ -218,7 +218,7 @@ void sched_maybe_preempt(void) {
   }
 }
 
-bool preempt_disabled(void) {
+__no_sanitize bool preempt_disabled(void) {
   thread_t *td = thread_self();
```

Comment:
Please consider adding a short comment explaining why `__no_sanitize` is required here.